packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1517,21 +1493,22 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     assert(cause != null);
     assert(from != null);
     _layoutText(minWidth: constraints.minWidth, maxWidth: constraints.maxWidth);
-    if (onSelectionChanged != null) {
-      final TextPosition firstPosition = _textPainter.getPositionForOffset(globalToLocal(from - _paintOffset));
-      final TextSelection firstWord = _selectWordAtOffset(firstPosition);
-      final TextSelection lastWord = to == null ?
-        firstWord : _selectWordAtOffset(_textPainter.getPositionForOffset(globalToLocal(to - _paintOffset)));
-
-      _handlePotentialSelectionChange(
-        TextSelection(
-          baseOffset: firstWord.base.offset,
-          extentOffset: lastWord.extent.offset,
-          affinity: firstWord.affinity,
-        ),
-        cause,
-      );
+    if (onSelectionChanged == null) {
```

Comment:
Why do all of these checks if onSelectionChanged exist here instead of one check inside of _handleSelectionChanged?

Reply:
I think the idea was to avoid doing the work needed to get to `_handleSelectionChanged`, but I think I'll move this up to _handleKeyEvent, and also a check in `_handleSelectionChanged`, since there are other call sites for it that aren't protected.

Reply:
Sounds good.